### PR TITLE
[Tooling] Reorg paths in promo screenshot lanes

### PR DESCRIPTION
**Pull request by AliSoftware**
This PR adds lots of small cleaning touches to the `screenshots.rb` lanes, especially the ones related to promo screenshots

### What

 - Remove `download_promo_strings` legacy lane and DRY everything in the more up-to-date `download_metadata_strings` lane
 - Extract out code that was doing some cleanup on the raw screenshot file names, for readability of main lane's code
 - Support separate promo config files for WordPress vs Jetpack, and add them to the `APP_SPECIFIC_VALUES` constant
 - Move the promotional copies from old location in `fastlane/playstoreres` to `fastlane/{jetpack_,}metadata/android/{locale}/…`
 - Move the promo screenshot assets from old location in `fastlane/playstoreres` to `fastlane/screenshots`.
 - Add the `Noto Serif` font to the repo, for easy install. A separate PR to the `release-toolkit` itself will add ability for the `promo_screenshots` action to check if fonts are installed, and prompt the user to install or activate them if missing or deactivated from FontBook.
 - Remove obsolete local legacy `fastlane/actions` and `fastlane/helpers`, now that we're using the ones from `release-toolkit` anyway

### To test

<details><summary>Known Issues, unrelated to this PR but impacting testing the PR</summary>

This is a bit tricky because despite those changes, running `create_promo_screenshots` will still fail and crash in the current state of things, but unrelated to the changes made from this PR… So the goal here will be to test that the changes made here work as expected, and that the crash we might get are only due to the the two following but unrelated things:

 - The config file references the image `1-create-a-site-or-start-a-blog.png` for one of its entry, but the current `WPScreenshotTest` UI tests don't generate such a raw screenshot. So the generation will crash when getting to the step trying to generate that image.
    -  One could remove the two entries in the `fastlane/screenshots/wordpress-config.json` config file (L70–76 and L136–142) before running the test. That should do the trick
    - But I didn't want to do that as part of this PR diff because I'm not sure if the right fix for this in the future will be to remove that entry… or to update the WPScreenshotTest to generate that raw screenshot again, so I'll let testers do that locally before testing
    - Once https://github.com/wordpress-mobile/release-toolkit/pull/429 lands and we start pointing to it, I might consider still keeping the entries about this `1-create-a-site-or-start-a-blog.png` file in the config files… but `# commenting` them (since YAML would now allow us to support that) so that we still have the config entry for it when we finally fix the UI tests to generate that raw screenshot again. Which is also why I kept it around for the time being.
 - There seems to be [a bug in `drawText` with Arabic text, making `drawText` crash when trying to generate the promo screenshots for that locale on an "Out of Bounds" exception](https://github.com/Automattic/drawText/issues/1).
    - One way to work around this in order to test this PR is to only test with a single latin-script locale, like `fr-FR`
</details>

#### Test Steps

 - Ensure you've installed everything needed to run the screenshot lanes, especially `brew install imagemagick` and `bundle install --with screenshots`
 - Run `bundle exec fastlane screenshots app:wordpress locale:fr-FR` to generate raw screenshots for French using both Pixel 3 and Nexus 9. Feel free to omit `locale:` parameter or to use a different comma-separated list of locales if you want to spice things up.
 - Edit the `fastlane/screenshots/wordpress-config.json` file and remove lines L70–76 and L136–142 — i.e. the entries containing the reference to `images/phoneScreenshots/1-create-a-site-or-start-a-blog.png` — to workaround the issue that this raw screenshot is missing (see "known issues" above for details)
 - Run `bundle exec fastlane create_promo_screenshots app:wordpress locale:fr-FR`.
    - You can add `force:true` parameter if you want to skip the confirmation step
    - You can add `skip_download_strings:true` parameter on subsequent calls to avoid re-downloading the latest string translations from GlotPress on every run
  - Check that the final images (2–7) have been generated in `fastlane/metadata/android/fr-FR/images/{phoneScreenshots,tenInchScreenshots}`

### CODE REVIEW

File: fastlane/lanes/localization.rb [hunk 1]
```diff
@@ -153,38 +153,48 @@
...
+  # Downloads the translated metadata (release notes, app store strings, title, etc.)
+  # from GlotPress, and updates the local `.txt` files with them.
+  #
+  # @option [String|Symbol] app The app to take screenshots for. Must be `wordpress` or `jetpack`
+  # @option [String] version The `versionName` of the app, used to extract the release notes from the right GlotPress key. Defaults to current `versionName`.
+  # @option [Int] build_number The `versionCode` of the app, used to save the release notes to the right `changelogs/*.txt` file. Defaults to current `versionCode`.
+  # @option [Boolean] skip_release_notes If set to true, will not download release notes. Defaults to `false`. This can be useful when all you want to download
+  #         is screenshots translations and metadata not linked to a specific version (in which case `version` and `build_number` parameters are optional).
+  # @option [Boolean] skip_commit If set to true, will skip the `git add`, `git commit` and `git push` operations. Default to false.
+  # @option [Boolean] skip_git_push If set to true, will skip the `git push` at the end. Default to false. Infered to `true` if `skip_commit` is `true`.
```
Comment (ParaskP7) [inline fastlane/lanes/localization.rb]: Minor (🔍): Typo `Infered` -> `Inferred`

File: fastlane/screenshots/jetpack-config.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
 	"version": 0.1,
 	"shadow_offset": 0,
 	"background_color": "#016087",
-	"stylesheet": "playstoreres/assets/style.css",
+	"stylesheet": "screenshots/assets/style.css",
 	"devices": [
 		{
 			"name": "Pixel 2 XL",
```
Comment (ParaskP7) [inline fastlane/screenshots/jetpack-config.json]: Suggestion (💡): Maybe we should take this opportunity and remove this `Pixel 2 XL` configuration? 🤔 

PS: The same apply for the `wordpress-config.json` file and such configuration.

**Review by ParaskP7 — APPROVED** [review]
👋 @AliSoftware !

I have reviewed and tested this PR as per the description, everything (almost) worked as expected. 💪 🥇 🌟 

Below, I am going to provide you with more details on what didn't go as I expected, that is, during me testing the solution, but I don't want to block this PR any further, thus I am going to approve it anyway.

-----

(1)

As I didn't have `imagemagick` installed, I went and installed it via `brew install imagemagick`. This installed this version for me: `Version: ImageMagick 7.1.0-52 Q16-HDRI aarch64 20549 https://imagemagick.org`

Afterwards, me trying to run `bundle install --with screenshots` didn't work. I ended up having this error: `An error occurred while installing rmagick (4.2.4), and Bundler cannot continue.`

Looking at the logs, I suspect that this is due to the fact that the needed `RMagick 4.2.4` installation/configuration is not working with Apple Silicon, see below:

```
======================================================================
Tue 29 Nov 22 12:30:15
This installation of RMagick 4.2.4 is configured for
Ruby 2.7.4 (x86_64-darwin21) and ImageMagick 7.1.0
======================================================================
```

As such, I couldn't complete the `create_promo_screenshots` tests, as per the test instructions: `bundle exec fastlane create_promo_screenshots app:wordpress locale:fr-FR`

Any thoughts on that? 🤔

-----

(2)

No matter, I was able to complete the `screenshots` tests, as per the test instructions: `bundle exec fastlane screenshots app:wordpress locale:fr-FR`

After a few tries, I also removed the lines `L70-76` and `L136-142` as the `images/phoneScreenshots/1-create-a-site-or-start-a-blog.png` wasn't being generated anyway.

However, what I noticed during this testing of mine was the following:

```
[11:07:34]: --------------------------------
[11:07:34]: --- Step: android_create_avd ---
[11:07:34]: --------------------------------
[11:07:37]: Installing System Image for Android 31 (system-images;android-31;google_apis;x86_64)
[11:07:37]: $ /Users/petros.paraskevopoulos/Library/Android/sdk/cmdline-tools/latest/bin/sdkmanager --install system-images\;android-31\;google_apis\;x86_64
[11:09:00]: ▸ [=======================================] 100% Unzipping... x86_64/data/misc/mod
[11:09:00]: System Image system-images;android-31;google_apis;x86_64 successfully installed.
[11:09:00]: Creating AVD `Pixel_3_API_31` (pixel_3, API 31)
[11:09:00]: $ /Users/petros.paraskevopoulos/Library/Android/sdk/cmdline-tools/latest/bin/avdmanager create avd --force --package system-images\;android-31\;google_apis\;x86_64 --device pixel_3 --sdcard 512M --name Pixel_3_API_31
[11:09:04]: ▸ [=======================================] 100% Fetch remote repository...
[11:09:04]: ▸ Auto-selecting single ABI x86_64
[11:09:05]: AVD `Pixel_3_API_31` successfully created.
```

From the above you will again notice the `x86_64` Intel architecture problem, this time, while I am trying to run the `screenshots` lane on Apple Silicon. I think this might create some additional problems later on, don't you think?

For example, it is downloading and installing the `system-images\;android-31\;google_apis\;x86_64` image. Normally, this is not allowed.

Is there a way we can switch to Apple Silicon, or was the Intel specific configuration on purpose? 🤔

-----

(3)

As per the test instructions, the 1st `images/phoneScreenshots/1-create-a-site-or-start-a-blog.png` image can't be generated and that's expected.
 
However, I am also not able to get this 7th `images/phoneScreenshots/7-the-worlds-most-popular-website-builder.png` image generated as well, I am only able to get 5 images in total (2-6).

Any thoughts on that? 🤔

-----

(4)

When running the `screenshots` for the Jetpack app via the `app:jetpack` option, it gets stuck during login. As such, this lane never completes and I needed to force kill it to proceed with the rest of my testing.
 
This is most probably due to the redesign of this screen, but I thought I should mention that anyway. Right? 🤔

**Review by mokagio — COMMENTED** [review]
This is most likely unrelated to the changes in this PR, but I wanted to log that when I tested with Jetpack, I got the following error:

```
java.lang.IllegalArgumentException: Zendesk needs to be setup before this method can be called
  ▸ 	at org.wordpress.android.support.ZendeskHelper.enablePushNotifications(ZendeskHelper.kt:197)
```

<details>
<summary>Details</summary>

```
[14:11:33]: ▸ java.lang.IllegalArgumentException: Zendesk needs to be setup before this method can be called
[14:11:33]: ▸ 	at org.wordpress.android.support.ZendeskHelper.enablePushNotifications(ZendeskHelper.kt:197)
[14:11:33]: ▸ 	at org.wordpress.android.push.GCMRegistrationIntentService.sendRegistrationToken(GCMRegistrationIntentService.java:86)
[14:11:33]: ▸ 	at org.wordpress.android.push.GCMRegistrationIntentService.lambda$onHandleWork$0$org-wordpress-android-push-GCMRegistrationIntentService(GCMRegistrationIntentService.java:53)
[14:11:33]: ▸ 	at org.wordpress.android.push.GCMRegistrationIntentService$$ExternalSyntheticLambda0.onComplete(Unknown Source:2)
[14:11:33]: ▸ 	at com.google.android.gms.tasks.zzi.run(com.google.android.gms:play-services-tasks@@18.0.1:1)
[14:11:33]: ▸ 	at android.os.Handler.handleCallback(Handler.java:938)
[14:11:33]: ▸ 	at android.os.Handler.dispatchMessage(Handler.java:99)
[14:11:33]: ▸ 	at android.os.Looper.loopOnce(Looper.java:201)
[14:11:33]: ▸ 	at android.os.Looper.loop(Looper.java:288)
[14:11:33]: ▸ 	at android.app.ActivityThread.main(ActivityThread.java:7839)
[14:11:33]: ▸ 	at java.lang.reflect.Method.invoke(Native Method)
[14:11:33]: ▸ 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
[14:11:33]: ▸ 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
[14:11:49]: ▸ INSTRUMENTATION_RESULT: shortMsg=Process crashed.
[14:11:49]: ▸ INSTRUMENTATION_CODE: 0
[14:11:49]: Pulling captured screenshots for locale it-IT from the device
[14:11:49]: $ /Users/gio/Library/Android/sdk/platform-tools/adb -s emulator-5554 shell echo \$EXTERNAL_STORAGE
[14:11:49]: ▸ /sdcard
[14:11:49]: $ /Users/gio/Library/Android/sdk/platform-tools/adb -s emulator-5554 shell echo \$ANDROID_DATA
[14:11:49]: ▸ /data
ls: /sdcard/com.jetpack.android/screengrab/it-IT/images/screenshots: No such file or directory
[14:11:49]: Exit status: 1
ls: /sdcard/Android/data/com.jetpack.android/files/screengrab/it-IT/images/screenshots: No such file or directory
[14:11:49]: Exit status: 1
ls: /data/user/0/com.jetpack.android/files/com.jetpack.android/screengrab/it-IT/images/screenshots: No such file or directory
[14:11:49]: Exit status: 1
ls: /data/data/com.jetpack.android/files/com.jetpack.android/screengrab/it-IT/images/screenshots: No such file or directory
[14:11:49]: Exit status: 1
ls: /data/data/com.jetpack.android/app_screengrab/it-IT/images/screenshots: No such file or directory
[14:11:49]: Exit status: 1
ls: /data/data/com.jetpack.android/screengrab/it-IT/images/screenshots: No such file or directory
[14:11:49]: Exit status: 1
[14:11:49]: Generating HTML Report
+-------------------------------------+-----------------------------------------------------------+
|                                          Lane Context                                           |
+-------------------------------------+-----------------------------------------------------------+
| DEFAULT_PLATFORM                    | android                                                   |
| PLATFORM_NAME                       | android                                                   |
| LANE_NAME                           | android screenshots                                       |
| GRADLE_ALL_APK_OUTPUT_PATHS         | ["/Users/gio/Developer/a8c/wpandroid/WordPress/build/out  |
|                                     | puts/apk/androidTest/jetpackVanilla/debug/org.wordpress.  |
|                                     | android-jetpack-vanilla-debug-androidTest.apk",           |
|                                     | "/Users/gio/Developer/a8c/wpandroid/WordPress/build/outp  |
|                                     | uts/apk/jetpackVanilla/debug/org.wordpress.android-jetpa  |
|                                     | ck-vanilla-debug.apk"]                                    |
| GRADLE_ALL_AAB_OUTPUT_PATHS         | []                                                        |
| GRADLE_ALL_OUTPUT_JSON_OUTPUT_PATHS | ["/Users/gio/Developer/a8c/wpandroid/WordPress/build/out  |
|                                     | puts/apk/androidTest/jetpackVanilla/debug/output-metadat  |
|                                     | a.json",                                                  |
|                                     | "/Users/gio/Developer/a8c/wpandroid/WordPress/build/outp  |
|                                     | uts/apk/jetpackVanilla/debug/output-metadata.json"]       |
| GRADLE_ALL_MAPPING_TXT_OUTPUT_PATHS | []                                                        |
| GRADLE_APK_OUTPUT_PATH              | /Users/gio/Developer/a8c/wpandroid/WordPress/build/outpu  |
|                                     | ts/apk/androidTest/jetpackVanilla/debug/org.wordpress.an  |
|                                     | droid-jetpack-vanilla-debug-androidTest.apk               |
| GRADLE_OUTPUT_JSON_OUTPUT_PATH      | /Users/gio/Developer/a8c/wpandroid/WordPress/build/outpu  |
|                                     | ts/apk/androidTest/jetpackVanilla/debug/output-metadata.  |
|                                     | json                                                      |
+-------------------------------------+-----------------------------------------------------------+
[14:11:49]: No such file or directory @ rb_sysopen - /Users/gio/Developer/a8c/wpandroid/fastlane/screenshots/jetpack/raw/screenshots.html
[14:11:49]: fastlane finished with errors
```



</details>

**Comment by AliSoftware** [comment]
Thank you both for the thorough reviews! ❤️ 

@ParaskP7 :

> Afterwards, me trying to run bundle install --with screenshots didn't work. I ended up having this error: An error occurred while installing rmagick (4.2.4), and Bundler cannot continue.

Yeah, I'll shove that under the "unfortunate quirks that can happen when trying to install `imagemagick` and `rmagick`" that I mention in [this documentation added in the other PR](https://github.com/wordpress-mobile/WordPress-Android/pull/17395/files#diff-7d89e1fae182d01a016ffdd13bbebc870060e1a49d899bd62ffef70dc5c0d92cR107).

> From the above you will again notice the x86_64 Intel architecture problem, this time, while I am trying to run the screenshots lane on Apple Silicon. I think this might create some additional problems later on, don't you think?
> 
> For example, it is downloading and installing the system-images\;android-31\;google_apis\;x86_64 image. Normally, this is not allowed.

This is strange, as [the logic is supposed to use `uname -m` to determine your Mac's architecture, then use `sdkmanager --list` to try and find a package name that contains said architecture](https://github.com/wordpress-mobile/release-toolkit/blob/06a77d8cd901ca204a623dc9c37f60358c3222a4/lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_emulator_helper.rb#L180-L183). And last time I tried on my M1 Mac—while testing [the PR that introduced this in `release-toolkit`](https://github.com/wordpress-mobile/release-toolkit/pull/409) some time ago—it worked as expected.

Can you tell me what are the output of `uname -m` and of `sdkmanager --list | grep 'system-images;android-.*;google_apis;arm64'`?

In any case, I think this is an issue that is outside of the scope of this PR (and is more related to [release-toolkit#409](https://github.com/wordpress-mobile/release-toolkit/pull/409)), so if this is confirmed depending on your answers of the above, you might want to open an issue on the release-toolkit repo, I guess.

> However, I am also not able to get this 7th images/phoneScreenshots/7-the-worlds-most-popular-website-builder.png image generated as well, I am only able to get 5 images in total (2-6).

Ah, indeed, I got the same here. Yet another case of a raw screenshot that is not generated by the current state of the UI Tests anymore and that would need fixing by Android devs at some point, just like the case for `1-create-a-site-or-start-a-blog.png`… 😒 

> When running the screenshots for the Jetpack app via the app:jetpack option, it gets stuck during login. As such, this lane never completes and I needed to force kill it to proceed with the rest of my testing.
>
> This is most probably due to the redesign of this screen, but I thought I should mention that anyway. Right? 🤔

Yes, it is. This is part of the work that I'm waiting on @AjeshRPai and @zwarm to find time to fix at some point, as I can't do much about it on my end. But other priorities seems to keep getting in their way so I'm not sure when an Android dev will find time to fix those Screenshot UI Tests.
See https://github.com/wordpress-mobile/WordPress-Android/issues/17128 and paaHJt-3wL-p2#project-is-now-on-hold for more details.

---

@mokagio 

> This is most likely unrelated to the changes in this PR, but I wanted to log that when I tested with Jetpack, I got the following error:
>
> ```
> java.lang.IllegalArgumentException: Zendesk needs to be setup before this method can be called
>  ▸ 	at org.wordpress.android.support.ZendeskHelper.enablePushNotifications(ZendeskHelper.kt:197)
> ```

Yes, this is a known issue. Basically, currently, the Jetpack Screenshot UI Tests are completely broken, due to the various bugs in the Android code that are tracked in https://github.com/wordpress-mobile/WordPress-Android/issues/17128 (+ due to login screen redesign as mentioned above by Petros), especially [this issue for the case of the ZenDesk crash](https://github.com/wordpress-mobile/WordPress-Android/issues/17130).
I've been asking for some Android devs to take a look and work on fixing those, but no one found the time so far, due to other things on their backlog constantly taking priority 😞 

So TL;DR this is a known issue (and why the testing instructions for this PR only included steps to test with WordPress, and not the crashing Jetpack case), and I'm not sure when someone will get to work on a fix, but it's unrelated to this PR indeed.

File: fastlane/screenshots/jetpack-config.json (hunk 2 above)
Comment (AliSoftware) [reply fastlane/screenshots/jetpack-config.json]: Good point, since it's not used in any of the subsequent `entries` anyway…

PS: Hopefully, once [release-toolkit#429](https://github.com/wordpress-mobile/release-toolkit/pull/429) gets approved and land, we will also be able to later convert the `{app}-config.json` config file into a `{app}-config.yaml` config file, which will allow us to start using YAML anchors (e.g. use `name: &phone "Pixel 3"` under `devices` declaration, then just use `device: *phone` in the entries, without having to replace the device name in all the places if/when we decide to change the device models again. But we'll need https://github.com/wordpress-mobile/release-toolkit/pull/429 for that first.

File: fastlane/lanes/localization.rb (hunk 1 above)
Comment (AliSoftware) [reply fastlane/lanes/localization.rb]: Addressed in 886cf1db1dea9e948676870b8b503d1f976a6a6a

File: fastlane/screenshots/jetpack-config.json (hunk 2 above)
Comment (AliSoftware) [reply fastlane/screenshots/jetpack-config.json]: Addressed in 443598b8b991cc4a19c4002d38d3e4bd04161a76

**Comment by AliSoftware** [comment]
> From the above you will again notice the x86_64 Intel architecture problem, this time, while I am trying to run the screenshots lane on Apple Silicon. I think this might create some additional problems later on, don't you think?
>
> For example, it is downloading and installing the system-images;android-31;google_apis;x86_64 image. Normally, this is not allowed.

Follow-up on that issue that @ParaskP7 was having here: this has now been resolved.

This was due to an issue in his setup where, despite running the command on an Apple Silicon Mac, and with its Terminal properly running in native mode (not Rosetta), his installation of Ruby was still `ruby 2.7.4p191 (2021-07-07 revision a21a3b7d23) [x86_64-darwin21]`—probably a leftover of the fact he did a standard migration (using Assistant Migration) from his Intel Mac when switching to his M1—hence why `uname -m` returned `arm64` as expected when run from his Terminal but returning `x86_64` when running from Ruby code.

The solution was for Petros to fix his installation of Ruby by reinstalling it properly (`brew upgrade rbenv ruby-build && rbenv uninstall 2.7.4 && rbenv install 2.7.4`) to get the ARM / M1 version instead of the Intel build.

File: fastlane/screenshots/jetpack-config.json (hunk 2 above)
Comment (ParaskP7) [reply fastlane/screenshots/jetpack-config.json]: Nice, thank you for addressing this @AliSoftware ! 🙇 

> PS: Hopefully, once https://github.com/wordpress-mobile/release-toolkit/pull/429 gets approved and land, we will also be able to later convert the {app}-config.json config file into a {app}-config.yaml config file, which will allow us to start using YAML anchors (e.g. use name: &phone "Pixel 3" under devices declaration, then just use device: *phone in the entries, without having to replace the device name in all the places if/when we decide to change the device models again. But we'll need https://github.com/wordpress-mobile/release-toolkit/pull/429 for that first.

Coolio! 🤞

**Comment by ParaskP7** [comment]
👋 @AliSoftware !

All in all, this is still a ✅ from my side, actually even more so, since now I am able to test all steps, excluding of course screenshots `1` and `7`. Everything is looking mergeable from my side.

> Yeah, I'll shove that under the "unfortunate quirks that can happen when trying to install imagemagick and rmagick" that I mention in [this documentation added in the other PR](https://github.com/wordpress-mobile/WordPress-Android/pull/17395/files#diff-7d89e1fae182d01a016ffdd13bbebc870060e1a49d899bd62ffef70dc5c0d92cR107).

As discussed internally, this is now resolved, it was a misconfiguration on my side, my `ruby` installation was pointing to `Intel`, see below:

```
ruby --version
 ruby 2.7.4p191 (2021-07-07 revision a21a3b7d23) [x86_64-darwin21]
```

After updating that, via `rbenv install 2.7.4`, now my `ruby` installation is pointing to `Apple` , see below:

```
ruby --version
 ruby 2.7.4p191 (2021-07-07 revision a21a3b7d23) [arm64-darwin22]
```

Now everything is working as expected. ✅ 

> This is strange, as [the logic is supposed to use uname -m to determine your Mac's architecture, then use sdkmanager --list to try and find a package name that contains said architecture](https://github.com/wordpress-mobile/release-toolkit/blob/06a77d8cd901ca204a623dc9c37f60358c3222a4/lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_emulator_helper.rb#L180-L183). And last time I tried on my M1 Mac—while testing https://github.com/wordpress-mobile/release-toolkit/pull/409 some time ago—it worked as expected.
> 
> Can you tell me what are the output of uname -m and of sdkmanager --list | grep 'system-images;android-.*;google_apis;arm64'?
> 
> In any case, I think this is an issue that is outside of the scope of this PR (and is more related to https://github.com/wordpress-mobile/release-toolkit/pull/409), so if this is confirmed depending on your answers of the above, you might want to open an issue on the release-toolkit repo, I guess.

Same as the above, it was a `ruby` misconfiguration on my side, which however is now resolved. ✅ 

> Ah, indeed, I got the same here. Yet another case of a raw screenshot that is not generated by the current state of the UI Tests anymore and that would need fixing by Android devs at some point, just like the case for 1-create-a-site-or-start-a-blog.png… 😒

Got it, I figured as much,  but thanks for the explanation! 👍 

PS: Double thanks for trying it out on your side as well. 🙏 

> Yes, it is. This is part of the work that I'm waiting on @AjeshRPai and @zwarm to find time to fix at some point, as I can't do much about it on my end. But other priorities seems to keep getting in their way so I'm not sure when an Android dev will find time to fix those Screenshot UI Tests.
> See https://github.com/wordpress-mobile/WordPress-Android/issues/17128 and paaHJt-3wL-p2#project-is-now-on-hold for more details.

Got it, I figured as much, but thanks for the explanation! 👍 

-----

Finally, and just as an FYI, with me not wanting to block this at all, just sharing, I now get the following error while running the `create_promo_screenshots` lane:

```
[11:35:31]: Exit status of command 'drawText html\=/Users/petros.paraskevopoulos/AndroidStudioProjects/automattic/WordPress-Android/fastlane/metadata/android/fr-FR/screenshot_6.txt maxWidth\=3000 maxHeight\=120 output\=/var/folders/xq/l24nqgw92x3cs25jw4jpt4bm0000gn/T/20221130-4731-mmr4c2 fontSize\=100px stylesheet\=./fastlane/screenshots/assets/style.css alignment\=center' was 1 instead of 0.
/Users/petros.paraskevopoulos/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.8/lib/bundler/rubygems_integration.rb:319:in `block in replace_bin_path': can't find executable drawText for gem fastlane-plugin-wpmreleasetoolkit (Gem::Exception)
	from /Users/petros.paraskevopoulos/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.8/lib/bundler/rubygems_integration.rb:347:in `block in replace_bin_path'
	from /Users/petros.paraskevopoulos/AndroidStudioProjects/automattic/WordPress-Android/vendor/bundle/ruby/2.7.0/bin/drawText:23:in `<main>'

...

[11:35:31]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

bundler: failed to load command: fastlane (/Users/petros.paraskevopoulos/AndroidStudioProjects/automattic/WordPress-Android/vendor/bundle/ruby/2.7.0/bin/fastlane)
Traceback (most recent call last):
	59: from /Users/petros.paraskevopoulos/.rbenv/versions/2.7.4/bin/bundle:23:in `<main>'
         ...
```

**Comment by AliSoftware** [comment]
> I now get the following error while running the create_promo_screenshots lane:

This is probably because (1) either you didn't install `drawText` — I should have added that step in the testing instructions, sorry… and in fact I should add it in [the new documentation](https://github.com/wordpress-mobile/WordPress-Android/pull/17395) — or (2) you have a former version of `drawText` still installed in your gem cache (because a while ago, `drawText` was bundled with the `release-toolkit` gem and installed in the `vendor/bundle` cache folder alongside it, now it's not anymore, but maybe you have some leftovers there). You might want to try `rm -rf vendor/bundle && bundle install` and try again.

The reason I'm thinking it's more likely to be (2) than (1) is that for the case (1) I'd expect us to catch the error and print a more explicit message outlining the instructions to install drawText. Which you might see after your retry if you didn't already install it using the new way since, btw.

**Comment by ParaskP7** [comment]
👋 @AliSoftware !

> This is probably because (1) either you didn't install drawText — I should have added that step in the testing instructions, sorry… and in fact I should add it in https://github.com/wordpress-mobile/WordPress-Android/pull/17395

👍 

> ...or (2) you have a former version of drawText still installed in your gem cache (because a while ago, drawText was bundled with the release-toolkit gem and installed in the vendor/bundle cache folder alongside it, now it's not anymore, but maybe you have some leftovers there). You might want to try rm -rf vendor/bundle && bundle install and try again.

Cool, thank you, as suggest I tried `rm -rf vendor/bundle && bundle install` first and then run the `create_promo_screenshots` lane again.

I got this first: `The rmagick gem doesn't seem to be installed. Be sure to use 'bundle install --with screenshots'.`

I then did `bundle install --with screenshots` again and then tried to run the `create_promo_screenshots` lane.

I go this then: `'drawText' not found – install it using 'brew install automattic/build-tools/drawText'.`

I then did `brew install automattic/build-tools/drawText` and then tried to run the `create_promo_screenshots` lane.

It now seems to work as expected! 🎉 

It is just that by the end I got this: `Unable to locate /Users/.../WordPress-Android/fastlane/screenshots/wordpress/raw_tmp/fr-FR/images/tenInchScreenshots/1-create-a-site-or-start-a-blog.png`

But, I guess this is expected due to the fact that we are not generating images `1` and `7`, right? 🤔

**Comment by AliSoftware** [comment]
> I got this first: The rmagick gem doesn't seem to be installed. Be sure to use 'bundle install --with screenshots'.

🤦  Of course. I should have said above `rm -rf vendor/bundle && bundle install --with screenshots` in the first place, not just `rm -rf vendor/bundle && bundle install`. My bad 😅 

> I go this then: `'drawText' not found – install it using 'brew install automattic/build-tools/drawText'.`

This is exactly the nicer error message that I was expecting for you to get (case (1)) 👍  So I'm glad this error-catching-and-nice-instructions-printing still works as expected for that 🙂 

I'll add that `brew install automattic/build-tools/drawText` to https://github.com/wordpress-mobile/WordPress-Android/pull/17395 explicitly right away.

> It now seems to work as expected! 🎉

🎉 

> It is just that by the end I got this: Unable to locate /Users/.../WordPress-Android/fastlane/screenshots/wordpress/raw_tmp/fr-FR/images/tenInchScreenshots/1-create-a-site-or-start-a-blog.png
>
> But, I guess this is expected due to the fact that we are not generating images 1 and 7, right? 🤔

Yep, exactly, just as mentioned in the PR description and as you guessed, this is indeed because the `wordpress-config.json` promo config file mentions images 1 and 7 but those are not generated by the current `JPScreenshotTest` Android instrumentation test anymore. We'll need to have and Android dev and/or designer decide if the right move is to keep them in the config file but fix the UI test to generate them again, or to get rid of them in the config file and stop generating promo screenshots for them. But that's for another PR, where we'd decide with Design and Marketing what the look for the next new promo screenshots should look like and what we need to generate, and that is unrelated to tooling fixes, so is outside of the scope of this PR.

**Comment by ParaskP7** [comment]
Great stuff @AliSoftware , thank you a lot for everything, and more so, for the thorough explanations, I think this is ready to merge now! 🥇  🚀 🎉
